src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -0,0 +1,234 @@
+    private final Supplier<Double> kDRight = rightPIDNamespace.addConstantDouble("kD", 0);
+    private final Supplier<Double> waitTimeRight = rightPIDNamespace.addConstantDouble("wait time", 0);
+    private final Supplier<Double> toleranceRight = rightPIDNamespace.addConstantDouble("tolerance", 0);
+    private final PIDSettings rightPIDSettings;
+
+    private final Namespace cameraPIDNamespace = namespace.addChild("camera pid");
+    private final Supplier<Double> kPCamera = cameraPIDNamespace.addConstantDouble("kP", 0);
+    private final Supplier<Double> kICamera = cameraPIDNamespace.addConstantDouble("kI", 0);
+    private final Supplier<Double> kDCamera = cameraPIDNamespace.addConstantDouble("kD", 0);
+    private final Supplier<Double> waitTimeCamera = cameraPIDNamespace.addConstantDouble("wait time", 0);
+    private final Supplier<Double> toleranceCamera = cameraPIDNamespace.addConstantDouble("tolerance", 0);
+    private final PIDSettings cameraPIDSettings;
+
+    private final Namespace feedForwardNamespace = namespace.addChild("feed forward");
+    private final Supplier<Double> kS = feedForwardNamespace.addConstantDouble("kS", 0);
+    private final Supplier<Double> kV = feedForwardNamespace.addConstantDouble("kV", 0);
+    private final Supplier<Double> kA = feedForwardNamespace.addConstantDouble("kA", 0);
+    private final FeedForwardSettings feedForwardSettings;
+
+    private final Namespace trapezoidProfileNamespace = namespace.addChild("trapezoid profile settings");
+    private final Supplier<Double> maxVelocity = trapezoidProfileNamespace.addConstantDouble("max velocity", 0);
+    private final Supplier<Double> trapezoidAcceleration = trapezoidProfileNamespace.addConstantDouble
+            ("acceleration", 0);
+    private final TrapezoidProfileSettings trapezoidProfileSettings;
+
+    public static Drivetrain getInstance() {
+        if (instance == null) {
+            instance = new Drivetrain(
+                    "drivetrain",
+                    new CANSparkMax(RobotMap.CAN.DRIVETRAIN_LEFT_SPARKMAX_MASTER,
+                            CANSparkMaxLowLevel.MotorType.kBrushless),
+                    new CANSparkMax(RobotMap.CAN.DRIVETRAIN_LEFT_SPARKMAX_SLAVE,
+                            CANSparkMaxLowLevel.MotorType.kBrushless),
+                    new CANSparkMax(RobotMap.CAN.DRIVETRAIN_RIGHT_SPARKMAX_MASTER,
+                            CANSparkMaxLowLevel.MotorType.kBrushless),
+                    new CANSparkMax(RobotMap.CAN.DRIVETRAIN_RIGHT_SPARKMAX_SLAVE,
+                            CANSparkMaxLowLevel.MotorType.kBrushless),
+                    SerialPort.Port.kUSB,
+                    TRACK_WIDTH);
+        }
+        return instance;
+    }
+
+    private Drivetrain(String namespaceName, CANSparkMax leftMaster, CANSparkMax leftSlave,
+                       CANSparkMax rightMaster, CANSparkMax rightSlave, SerialPort.Port gyroPort, double trackWidth) {
```

Comment:
Give the AHRS as a parameter, not the port. It's like giving the spark max port as a parameter